mne/viz/utils.py

```diff
@@ -36,9 +36,9 @@
                          _divide_to_regions)
 from ..annotations import Annotations, _sync_onset
 
-
-COLORS = ['b', 'g', 'r', 'c', 'm', 'y', 'k', '#473C8B', '#458B74',
-          '#CD7F32', '#FF4040', '#ADFF2F', '#8E2323', '#FF1493']
+BAD_COLOR = u'#d62728'
+COLORS = [u'#1f77b4', u'#ff7f0e', u'#2ca02c', BAD_COLOR, u'#9467bd',
+          u'#8c564b', u'#e377c2', u'#7f7f7f', u'#bcbd22', u'#17becf']
```

Comment:
I would not do it like this. If we now use mpl defaults we should use its rcParams. If someone want to change the MNE colors it would be able to do it via setting the global mpl default values.

Reply:
but how would you identify a "bad color" out of the rcParams? Especially, if these colors could change anytime via the global mpl default vals?

Reply:
1. Use whatever is in `rcParams` at the time of plotting (so we'll need a private `_get_color_cycle` function or so that we call whenever we start a plotting function that wants a color cycle)
2. For annotations (this logic can also be in `_get_color_cycle(..., annotations=False)`):
    1. If `'r'` is in the list, use it as bad color when plotting annotations
    2. If not, add it and use it